api/V1/Recording.js

```diff
@@ -0,0 +1,201 @@
+          });
+          recording.set('rawFileKey', key);
+          recording.set('DeviceId', device.id);
+          recording.set('GroupId', device.GroupId);
+          recording.set('processingState',
+            models.Recording.processingStates[data.type][0]);
+          if (typeof device.public === 'boolean')
+            recording.set('public', device.public);
+          validData = true;
+        } catch (e) {
+          log.debug(e);
+          // TODO Stop stream here.
+        }
+      });
+
+      // Stream file to LeoFS.
+      form.on('part', (part) => {
+        if (part.name != 'file') return part.resume();
+        validFile = true;
+        log.debug('Streaming file to LeoFS.')
+        var s3 = new AWS.S3({
+          endpoint: config.leoFS.endpoint,
+          accessKeyId: config.leoFS.publicKey,
+          secretAccessKey: config.leoFS.privateKey,
+        });
+        s3.upload({
+          Bucket: config.leoFS.bucket,
+          Key: key,
+          Body: part,
+        }, (err, data) => {
+          if (err) return log.error(err);
+          log.info("Finished streaming file to LeoFS Key:", key);
+        });
+      });
+
+      // Close response.
+      form.on('close', async () => {
+        log.info("Finished POST request.");
+        if (!validFile || !validData)
+          return responseUtil.invalidDatapointUpload(response);
+        await recording.validate();
+        await recording.save();
+        return responseUtil.validDatapointUpload(response);
+      });
+
+      form.on('error', (e) => {
+        log.error(e);
+      })
+
+      form.parse(request);
+    });
+
+  /**
+   * @api {get} /api/v1/recordings Query available recordings
+   * @apiName GetRecordigns
+   * @apiGroup Recordings
+   *
+   * @apiUse V1UserAuthorizationHeader
+   *
+   * @apiParam {JSON} where [Sequelize where conditions](http://docs.sequelizejs.com/manual/tutorial/querying.html#where) for query
+   * @apiParam {Number} offset Query result offset (for paging).
+   * @apiParam {Number} limit Query result limit (for paging).
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiSuccess {Number} offset Mirrors request offset parameter.
+   * @apiSuccess {Number} limit Mirrors request limit parameter.
+   * @apiSuccess {Number} count Total number of records which match the query.
+   * @apiSuccess {JSON} rows List of details for records which matched the query.
+   *
+   * @apiUse V1ResponseError
+   */
+  app.get(
+    apiUrl,
+    passport.authenticate(['jwt'], { session: false }),
+    async (request, response) => {
+      log.info(request.method + " Request: " + request.url);
+
+      // Check that the request was authenticated by a User.
+      if (request.user !== null && !requestUtil.isFromAUser(request))
+        return responseUtil.notFromAUser(response);
+
+      var where = request.query.where;
+      var offset = Math.floor(parseInt(request.query.offset));
+      var limit = Math.floor(parseInt(request.query.limit));
```

Comment:
Why is `Math.floor` required here? Isn't the output of `parseInt` already an integer value?

Reply:
Yeah, you are right. Not too sure what I was doing..